src/main/java/us/springett/nistdatamirror/NistDataMirror.java

```diff
@@ -93,7 +93,7 @@ public NistDataMirror(String outputDirPath, String type) {
     private Proxy initProxy() {
         String proxyHost = System.getProperty("http.proxyHost");
         String proxyPort = System.getProperty("http.proxyPort");
-        if (proxyHost != null && proxyPort != null) {
+        if (proxyHost != null && !"".equals(proxyHost) && proxyPort != null && !"".equals(proxyPort)) {
```

Comment:
I suggest to use

```java
if (proxyHost != null && !proxyHost.trim().isEmpty() && proxyPort != null && !proxyPort.trim().isEmpty()) {
```

or 

```java
if (proxyHost != null && !proxyHost.isBlank() && proxyPort != null && !proxyPort.isBlank()) {
```
if Java 11+ can be used.

Reply:
Improvement committed.